docs/setup.md

```diff
@@ -100,7 +100,120 @@ in their associated documentation (chapter "Components").
+
+Make sure to run `varialbes.sh` if you set a custom `BC_CONFIG` variables. If
+set, this variable tells `bcadmin` where to store and get the configuration
+files.
+
+Create a new skipchain:
+
+```bash
+# This file was created by the run_nodes.sh script
+bcadmin create ledger/conode/tmp/public.toml 
+# The output of the command offers you to export the BC variable. Copy/past the
+# last line into to your terminal
+export BC="path/to/folder/bc-BYZCOIN_ID.cfg"
+```
+
+It is also a good idead to save the `export BC=...` command in the
+`variables.sh`.
+
+Upon creation, bcadmin created for you the admin identity and DARC, which are
+described by the config file saved in the BC variable. You can display the
+content of this configuration file with the following command:
+
+```bash
+bcadmin info
+```
+
+#### Initialize Calypso
+
+In the following steps we will set up Calypso. Calypso is the technology that
+allows secrets sharing on the blockchain. In our case, we use it to store the
+symmetric keys of the encrypted datasets. The steps are performed with the
+`csadmin` command line utility. If you followed the steps in "Generate the
+executables", then the `csadmin` command line utility should already be in your
+path.
+
+The first step is to authorize each node in participating to a secret sharing
+protocol. You must run one command for each server, giving the path to its
+private key file and the Byzcoin ID. The Byzcoin ID can be retreived with the
+`bcadmin info` command:
+
+```bash
+# Display the configuration info: spot the Byzcoin ID
+bcadmin info
+# Authorize each node:
+csadmin authorize ledger/conode/tmp/co1/private.toml BYZCOIN_ID
+csadmin authorize ledger/conode/tmp/co2/private.toml BYZCOIN_ID
+csadmin authorize ledger/conode/tmp/co3/private.toml BYZCOIN_ID
+```
+
+The next step is to setup a long term secret and then launch a distributed key
+generation protocol. Do those two steps with the following command. You will
+need the output of the first command to launch the second one:
+
+```bash
+# setup a long term secret, note the instance id
+csadmin contract lts spawn
+# start a distributed key generation protocol
+csadmin dkg start -i INSTANCE_ID
+```
+
+The instance id that outputs the first command correponds to the needed `LtsID`
+configuration entry in `domanager/app/config.toml`, while the "X" that is
+displayed by the second command corresponds to the `LtsKey` entry.
+
+#### Add rules to the DARC
+
+In order to use the custom Odyssey smart contracts, we need to allow actions on
+those smart contract in our DARC (the one specified by the BC config file).
+
+Launch the following commands or only those containing the rules you are
+interested in:
+
+```bash
+# Firstly, spot and note your identity (something like "ed25519:aef123...")
+bcadmin info
+# Save the identity into a variable
+id=IDENTITY
+# Note: do not run those commands into one single batch in order to let the 
```

Comment:
How about fixing `bcadmin darc rule` to take multiple rules and put them into one transaction? Sorry, I know it's kind of boring work, but...

Reply:
then we should take a configuration file as input and parse it, as we want (not in this particular case though) to specify N rules that can have each K different rules. Not impossible but requires some work in order to speed up a set up process that should happen only once.

This is not the first time we are limited with the edition of DARCs. There is also this story where we cannot edit a single rule but only replace it entirely. A "DARC editor" could be a funny semester project.